nibabel/spatialimages.py

```diff
@@ -349,21 +416,21 @@ def check_slicing(self, slicer, return_spatial=False):
             Validated slicer object that will slice image's `dataobj`
             without collapsing spatial dimensions
         """
-        slicer = canonical_slicers(slicer, self.img.shape)
+        canonical = canonical_slicers(slicer, self.img.shape)
         # We can get away with this because we've checked the image's
         # first three axes are spatial.
         # More general slicers will need to be smarter, here.
-        spatial_slices = slicer[:3]
+        spatial_slices = canonical[:3]
         for subslicer in spatial_slices:
             if subslicer is None:
                 raise IndexError('New axis not permitted in spatial dimensions')
             elif isinstance(subslicer, int):
                 raise IndexError(
                     'Scalar indices disallowed in spatial dimensions; Use `[x]` or `x:x+1`.'
                 )
-        return spatial_slices if return_spatial else slicer
+        return spatial_slices if return_spatial else canonical
 
-    def slice_affine(self, slicer):
+    def slice_affine(self, slicer: object) -> np.ndarray:
```

Comment:
Again, I'm probably missing something, but just making sure `object` is the best choice here.

Reply:
Here the point is that you really can pass anything to this. We narrow the type by using `check_slicing`, which narrows it by using `canonical_slicers`. If it would be helpful, I have some updates to `flieslice` (where `canonical_slicers` is defined). I didn't want to muddy this PR, but I can pull those in if you like.

Reply:
No need, thank you. Just wanted to double-check with you (and learn).